1.27/Dockerfile

```diff
@@ -66,14 +67,34 @@ RUN set -eux; \
 
 # Version
 ENV MEDIAWIKI_MAJOR_VERSION 1.27
-ENV MEDIAWIKI_BRANCH REL1_27
 ENV MEDIAWIKI_VERSION 1.27.7
-ENV MEDIAWIKI_SHA512 3dcee01d483be27eed1a901ef24a9080a2cccf3d9ee4eebc544d0c0fdb5a1a7272a68daa024d89754542e56a4d19eebec407775b38a00f067f2179f712deda76
 
 # MediaWiki setup
 RUN set -eux; \
+	fetchDeps=" \
+		gnupg \
+		dirmngr \
+	"; \
+	apt-get update; \
+	apt-get install -y --no-install-recommends $fetchDeps; \
+	\
 	curl -fSL "https://releases.wikimedia.org/mediawiki/${MEDIAWIKI_MAJOR_VERSION}/mediawiki-${MEDIAWIKI_VERSION}.tar.gz" -o mediawiki.tar.gz; \
-	echo "${MEDIAWIKI_SHA512} *mediawiki.tar.gz" | sha512sum -c -; \
+	curl -fSL "https://releases.wikimedia.org/mediawiki/${MEDIAWIKI_MAJOR_VERSION}/mediawiki-${MEDIAWIKI_VERSION}.tar.gz.sig" -o mediawiki.tar.gz.sig; \
+	export GNUPGHOME="$(mktemp -d)"; \
+# gpg key from https://www.mediawiki.org/keys/keys.txt
```

Comment:
Any reason not to simply import https://www.mediawiki.org/keys/keys.txt ? The SKS keyserver infrastructure is no longer reliable/safe to use.

Reply:
You should definitely consider https://keys.openpgp.org/ (see https://keys.openpgp.org/about for more details)

It's a lot more reliable, and the design makes it resilient to the kind of attacks getting more popular on SKS recently.

See https://github.com/docker-library/faq#openpgp--gnupg-keys-and-verification for a little more detail on why we're not keen on simply importing keys files directly.

Reply:
Keyservers can take revocations into account and are decentralized.
Why are they insecure? As long as the full fingerprint is used it's still incredible hard to find collisions.

Reply:
I believe https://dkg.fifthhorseman.net/blog/openpgp-certificate-flooding.html / https://dkg.fifthhorseman.net/blog/community-impact-openpgp-cert-flooding.html is what @legoktm is referring to.

Reply:
IMHO Hagrid has other drawbacks, but feel free to replace this with you favorite keyserver